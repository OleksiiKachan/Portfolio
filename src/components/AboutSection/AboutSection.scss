@import '../../assets/style/variables';

.aboutSection {
  &__introduction {
    margin: 3rem;
  }

  &__introductionParagraph {
    margin: 1.5rem 0;
    text-align: justify;
  }

  &__rightColumn {
    margin: 3rem;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: flex-end;
  }

  &__availabilityLabel {
    font-size: 1.8rem;
  }

  &__availabilityDate {
    font-size: 1.6rem;
    font-style: italic;
  }

  &__titleText {
    display: none;
  }
}

@media screen and (max-width: $breakpoint-tablet) {
  .aboutSection {
    &__titleText {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: flex-end;
      padding: 2rem 1rem;
      text-align: right;
    }
    &__title {
      font-size: 1.2rem;
      margin: 0.5rem 0;
    }

    &__subtitle {
      font-size: 1rem;
    }

    &__resumeLink {
      color: $color-link;
      text-decoration: none;
      display: flex;
      align-items: flex-end;
      font-size: 0.8rem;
      font-weight: 700;
      margin: 1rem 0;

      &::after {
        content: '';
        width: 12px;
        height: 12px;
        margin-left: 4px;
        background-size: contain;
        background-repeat: no-repeat;
        background-image: url('../../assets/icons/icon_externalResource.svg');
      }
    }

    &__introduction {
      margin: 0 3rem;
    }

    &__availabilityLabel {
      font-size: 1.4rem;
    }
  }
}
