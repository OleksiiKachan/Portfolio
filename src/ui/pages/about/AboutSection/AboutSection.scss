@import '../../../assets/style/variables';
@import '../../../assets/style/mixins';

.aboutSection {
  &__title {
    @include respond(tablet) {
      margin: 0.5rem 0;
    }
  }

  &__subtitle {
    @include respond(tablet) {
    }
  }

  &__introduction {
    margin: 3rem;

    @include respond(tablet) {
      margin: 0 3rem;
    }
  }

  &__introductionParagraph {
    margin: 1.5rem 0;
    text-align: justify;
  }

  &__rightColumn {
    margin: 3rem;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: flex-end;
  }

  &__availabilityLabel {
    font-size: 1.8rem;

    @include respond(tablet) {
      font-size: 1.4rem;
    }
  }

  &__availabilityDate {
    font-size: 1.6rem;
    font-style: italic;
  }

  &__titleText {
    display: none;

    @include respond(tablet) {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: flex-end;
      padding: 2rem 1rem;
      text-align: right;
    }
  }

  &__resumeLink {
    @include respond(tablet) {
      color: $color-link;
      text-decoration: none;
      display: flex;
      align-items: flex-end;
      font-size: 0.8rem;
      font-weight: 700;
      margin: 1rem 0;

      &::after {
        content: '';
        width: 12px;
        height: 12px;
        margin-left: 4px;
        background-size: contain;
        background-repeat: no-repeat;
        background-image: url('../../../assets/icons/icon_externalResource.svg');
      }
    }
  }
}
