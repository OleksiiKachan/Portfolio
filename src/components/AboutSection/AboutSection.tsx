import React from 'react';
import classNames from 'classnames';
import './AboutSection.scss';
import CoverImage from './CoverImage';

/*
    <ContactSection contacts={} className='' />
*/

type PropsType = {
  aboutInfo: {
    name: string;
    title: string;
    introduction: Array<string>;
    smContacts: Array<{ type: string; url: string }>;
    resumeUrl: string;
    availabilityDate: string;
  };
  className?: string;
};

export default ({ aboutInfo, className }: PropsType) => {
  return (
    <div id="about" className={classNames('aboutSection', className)}>
      <CoverImage
        name={aboutInfo.name}
        title={aboutInfo.title}
        resumeUrl={aboutInfo.resumeUrl}
        contacts={aboutInfo.smContacts}
      />
      <div className={classNames('aboutSection__introduction', className)}>
        {aboutInfo.introduction.map(item => {
          return (
            <p
              key={aboutInfo.introduction.indexOf(item)}
              className={classNames('aboutSection__introductionParagraph')}
            >
              {item}
            </p>
          );
        })}
      </div>
      <div className={classNames('aboutSection__rightColumn')}>
        <p className={classNames('aboutSection__availabilityLabel')}>
          Available for work:{' '}
          <span className={classNames('aboutSection__availabilityDate')}>
            {aboutInfo.availabilityDate}
          </span>
        </p>
      </div>
    </div>
  );
};
