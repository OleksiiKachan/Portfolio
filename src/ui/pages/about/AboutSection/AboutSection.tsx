import React from 'react';
import classNames from 'classnames';
import './AboutSection.scss';
import CoverImage from '../CoverImage/CoverImage';
import { ParsedText } from '../../../elements';

/*
    <AboutSection
      aboutInfo={}
      className=''
    />
*/

type PropsType = {
  aboutInfo: {
    name: string;
    title: string;
    introduction: Array<string>;
    smContacts: Array<{ type: string; url: string }>;
    resumeUrl: string;
    availabilityDate: string;
    portfolioStack: Array<string>;
    repository: string;
    coverImage: string;
  };
  className?: string;
};

export default ({ aboutInfo, className }: PropsType) => {
  return (
    <div id="about" className={classNames('aboutSection', className)}>
      <CoverImage
        name={aboutInfo.name}
        title={aboutInfo.title}
        resumeUrl={aboutInfo.resumeUrl}
        contacts={aboutInfo.smContacts}
        coverImage={aboutInfo.coverImage}
      />

      <div className={classNames('aboutSection__titleText')}>
        <h1 className={classNames('aboutSection__title')}>{aboutInfo.name}</h1>
        <h2 className={classNames('aboutSection__subtitle')}>
          {aboutInfo.title}
        </h2>
        <a
          href={aboutInfo.resumeUrl}
          target="_blank"
          rel="noopener noreferrer"
          className={classNames('aboutSection__resumeLink')}
        >
          Printable CV
        </a>
      </div>
      <div className={classNames('aboutSection__introduction')}>
        {aboutInfo.introduction.map(item => {
          return (
            <ParsedText
              key={aboutInfo.introduction.indexOf(item)}
              className={classNames('aboutSection__introductionParagraph')}
            >
              {item}
            </ParsedText>
          );
        })}
      </div>
      <div className={classNames('aboutSection__rightColumn')}>
        <p className={classNames('aboutSection__availabilityLabel')}>
          Available for work:{' '}
          <span className={classNames('aboutSection__availabilityDate')}>
            {aboutInfo.availabilityDate}
          </span>
        </p>
      </div>
      <div className={classNames('aboutSection__portfolioStack')}>
        <p>This site is created using:</p>
        <p>{aboutInfo.portfolioStack.join(', ')}</p>
        <a
          href={aboutInfo.repository}
          target="_blank"
          rel="noopener noreferrer"
          className={classNames('aboutSection__portfolioRepository')}
        >
          {aboutInfo.repository}
        </a>
      </div>
    </div>
  );
};
