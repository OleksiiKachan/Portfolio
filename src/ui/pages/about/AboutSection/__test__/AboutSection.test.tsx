import React from 'react';
import { shallow } from 'enzyme';
import AboutSection from '../AboutSection';

describe('AboutSection', () => {
  it('Snapshot', () => {
    const component = shallow(
      <AboutSection
        aboutInfo={{
          name: 'name',
          title: 'title',
          introduction: ['introduction string 1', 'introduction string 2'],
          smContacts: [
            { type: 'type1', url: 'url1' },
            { type: 'type2', url: 'url2' },
          ],
          resumeUrl: 'resume.url',
          availabilityDate: 'availability information',
          portfolioStack: ['item1', 'item2'],
        }}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
