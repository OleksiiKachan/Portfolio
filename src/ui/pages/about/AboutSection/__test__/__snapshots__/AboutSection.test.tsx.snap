// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutSection Snapshot 1`] = `
<div
  className="aboutSection"
  id="about"
>
  <_default
    contacts={
      Array [
        Object {
          "type": "type1",
          "url": "url1",
        },
        Object {
          "type": "type2",
          "url": "url2",
        },
      ]
    }
    name="name"
    resumeUrl="resume.url"
    title="title"
  />
  <div
    className="aboutSection__titleText"
  >
    <h1
      className="aboutSection__title"
    >
      name
    </h1>
    <h2
      className="aboutSection__subtitle"
    >
      title
    </h2>
    <a
      className="aboutSection__resumeLink"
      href="resume.url"
      rel="noopener noreferrer"
      target="_blank"
    >
      Printable CV
    </a>
  </div>
  <div
    className="aboutSection__introduction"
  >
    <_default
      className="aboutSection__introductionParagraph"
      key="0"
    >
      introduction string 1
    </_default>
    <_default
      className="aboutSection__introductionParagraph"
      key="1"
    >
      introduction string 2
    </_default>
  </div>
  <div
    className="aboutSection__rightColumn"
  >
    <p
      className="aboutSection__availabilityLabel"
    >
      Available for work:
       
      <span
        className="aboutSection__availabilityDate"
      >
        availability information
      </span>
    </p>
  </div>
  <div
    className="aboutSection__portfolioStack"
  >
    <p>
      This site is created using:
    </p>
    <p>
      item1, item2
    </p>
    <a
      className="aboutSection__portfolioRepository"
      href="project repo"
      rel="noopener noreferrer"
      target="_blank"
    >
      project repo
    </a>
  </div>
</div>
`;
